Greg

gpric024 opened the pull request:
Made changes and new test class based on previous feedback

gpric024 committed:
Uploading original results notebook

gpric024 committed:
adding the injected errors process

gpric024 committed:
adding class files and dicts

gpric024 committed:
removing old files

gpric024 committed:
how to get the datasets

gpric024 committed:
Update DataLoader.py

adding encoding commit suggestion

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

Same as last commit

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

Same as last two commits

Co-authored-by: Shaomei <shaomei@gmail.com>

gpric024 committed:
Update DataLoader.py

added doc string

gpric024 committed:
Update DyslexiaInjector.py

Updated with Shaomei's comments in mind, still need to address capitalization and nested loops for p_homophone and p_letter

gpric024 committed:
Update DyslexiaInjector.py

Modified code so we now include capitalized words and words with punctuation in our swapping corpus

gpric024 committed:
Update DyslexiaInjector.py

Just updated some comments

gpric024 committed:
Updates

Made required changed to dyslexia injector and data loader. Bug fixes and slightly optimized.

gpric024 committed:
Made changes to clean up code and testing of the injector class

gpric024 committed:
Added Pedler's list

gpric024 committed:
Added testing case for confusing word

gpric024 committed:
For dev

gpric024 committed:
Minor fixes nad pushing to work on another computer

gpric024 committed:
Added broader sentence level test

This method essentially tests the entire injection process

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,311 @@
+        Injects dyslexia into a sentence with a given probability
+    """
+    def __init__(self, load: DataLoader, homophone_path = "data/homophones_dict.pickle",
+                confusing_letters_path = "data/confusing_letters_dict.pickle", confusing_words_path="data/pedler_dict.pickle", seed = 42):
```
nit: line too long, try to keep line length within 80 characters for better readability.

Similar for others long lines below.

Shaomei commented on DataLoader.py:
```diff
@@ -0,0 +1,112 @@
+    create_deepcopy():
+        Returns a deep copy of the class instance
+    get_name():
+        returns name of the dataset (dataset_name)
```
Can you add a section for "Usage", and show a few examples of how to use this class?

Shaomei commented on DyslexiaInjector.py:
```diff
@@ -0,0 +1,311 @@
+        Returns the confusing letters of a letter
+    injector(sentence, p_homophone, p_letter)
+        Injects dyslexia into a sentence with a given probability
+    """
```
Same as above. This class is getting quite complicated, could be helpful to provide a few examples of how to use it here.

Shaomei commented on TestInjector.py:
```diff
@@ -0,0 +1,115 @@
+        #check that first letter is capitalized
+        self.assertEqual(out_word[0].isupper(), True)
+        #check that words are different
+        self.assertNotEqual(out_word, word)
```
What do we expect the out_word to be? Can we use another assert specify it here, too?

Shaomei commented on TestInjector.py:
```diff
@@ -0,0 +1,115 @@
+            if out_word[i].lower() != word[i].lower():
+                swapped_count += 1
+        #checking if correct number of letters were swapped
+        self.assertEqual(swapped_count, letters_swapped, f"actual_swapped_count: {swapped_count} | given_letters_swapped: {letters_swapped}")
```
Besides checking letters_swapped, can we also check whether the out_word is what we expected to be?

Shaomei commented on TestInjector.py:
```diff
@@ -0,0 +1,115 @@
+        word = "back"
+        out_word = self.injector.confusing_word_injector(original_word, word)
+        print(f"original word: {original_word} | out_word: {out_word}")
+        self.assertNotEqual(out_word, word)
```
So what is the expected out_word?

gpric024 replied:
added regex to get the proper output

Shaomei commented on TestSentence.py:
```diff
@@ -0,0 +1,34 @@
+
+import unittest
+#objective of this test is to make sure that the sentences in the dataset are clean
```
Hmm, checking the sentences in the dataset does not sound like a unit test. 

Maybe we should just make it into a PrecheckDataset class and tell people (including ourselves) to run it before the injection.

gpric024 replied:
I do have a fix_formatting method in the dataloader class which handles all these cases. This is more to ensure that nothing has slipped through the cracks.  For now, I think I can remove it.

Shaomei reviewed (approved):
Looking good overall. 

You can merge after addressing my comments, no need to review again.

gpric024 committed:
Making changes based on Shaomei's comments

Added usage examples. shortened some lines of code, added more assertions to some test cases
